kernel/kernel-api/src/main/java/io/delta/kernel/internal/stats/FileSizeHistogram.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.stats;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.*;
+
+/** A histogram that tracks file size distributions and their counts. */
```

Comment:
Can you talk about what the public API will be? i.e. you can insert and remove ... but that is just modification ... what do you _do_ with it once you have modified it? will it just be `toRow`?

Reply:
Currently we will provide fromColumnVector() and toRow(). The potential use case(CRC) only need these